### DISCUSSION TITLE
Numerical underflow warning

### DISCUSSION THREAD

**Discussion by cblair8181** [iqtree/iqtree2#365] [Q&A] [answered]
Dear colleagues, 

I am running an IQ-TREE analysis with >7 million bp for 83 individuals. Unfortunately, I am running into some issues at the model selection stage. I am receiving the following repeated messages:

Pattern xxxx likelihood is underflown

WARNING: Numerical underflow for lh-branch

Here is the command I used.

Command: ./iqtree2 -s MVZ228761_pruned.phy --alrt 1000 -B 1000 -T AUTO -safe

Note that I tried both with and without the -safe flag.

I am using version 2.3.6 for Linux. Any help would be greatly appreciated. Thanks.

Chris

**Comment by roblanf** [comment]
The issue is probably that the huge alignments creates *very* small numbers. The best thing to try is to add:

`-safe`

to your command line. This will attempt to avoid numerical underflow, but with such a large alignment it may still not work. Can you try it and let us know how it goes? If you still get warnings, then sending the dataset would be useful.

Rob

**Reply by cblair8181** [reply]
Hi Rob, 

Thank you for your reply. Please note that I receive these warnings both with and without the -safe flag. 

Chris

**Comment by cblair8181** [comment]
Hi all, 

This analysis eventually finished, but I am not sure if the results are trustworthy. 


Chris

**Comment by roblanf** [comment, marked as answer]
Just adding here an answer that @thomaskf sent me:

Using the `--safe` option is definitely better than not using it. However, if the warning still appears after using this option, it’s okay to ignore it. Within IQ-TREE (specifically in the `phylokernelnew.h` file), there’s a built-in procedure to handle numerical underflow issues. If the issue cannot be resolved, IQ-TREE will fail to compute a valid final likelihood value and will terminate.
Since, in this case, IQ-TREE only shows a warning and continues running, there is no need to worry. The warnings can be ignored. 

So - fine to continue and interpret!

**Reply by cblair8181** [reply]
Thank you!
